bin/mail/2016-03-29-ny-cs4all/generate-list.rb

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env ruby
+require_relative '../mailing-common/mailing-list-utils'
+require 'set'
+
+# email addresses *@schools.nyc.gov who are (any of):
+# a- Code Studio teachers who have already been through the K-5 PD
+# b- Code Studio teachers who have not been through any PD
+# c- Hour of Code teachers who aren't on Code Studio.
+
+def query_dashboard_teachers(query)
+  {}.tap do |results|
+    DASHBOARD_DB.fetch(query).each do |teacher|
+      email = teacher[:email]
+      results[email] = {email: email, name: teacher[:name], id: teacher[:id]}
+    end
+  end
+end
+
+# a- Code Studio teachers who have already been through the K-5 PD
+SECTION_IDS = Set.new
+SOLR_PARAMS = {
+  q: 'kind_s:ProfessionalDevelopmentWorkshop',
+  fl: 'section_id_s',
+  rows: 10000
+}
+SOLR.query(SOLR_PARAMS).each do |record|
+  next if record['section_id_s'].to_s == ''
+  SECTION_IDS << record['section_id_s']
+end
+puts "#{SECTION_IDS.count} K5 workshop section Ids loaded from solr"
+
+k5_pd_teachers = query_dashboard_teachers <<eos
+  SELECT DISTINCT culled_users.id, culled_users.email, culled_users.name
+  FROM (
+    SELECT * FROM users WHERE email like '%@schools.nyc.gov'
+  ) AS culled_users
+  JOIN followers ON (followers.student_user_id = culled_users.id)
```

Comment:
student_user_id. This is looking for teachers what attended the workshop. Followers.user_id is the workshop facilitator.